### PR TITLE
Saga into master

### PR DESCRIPTION
**Pull request by plantysnake**

### CODE REVIEW

**Comment by marina-igorevna-mishur** [comment]
1. would make sense to have just one `tableSaga`
2. would make sense to have a separate `tableReducer`
3. no reason to remove the service
4. why change standard `mapStateToProps `, `mapDispatchToProps` etc. names?
5. please make full use of actions, use them to dispatch from component and saga and store all string consts for types
6. connect will dispatch automatically, see https://blog.benestudio.co/5-ways-to-connect-redux-actions-50e4ad5ea096 #2

_*bonus_: add loading indicator to `App` and loading state to `rootReducer` (might need to set a timeout to actually see it)

**Comment by plantysnake** [comment]
- [ x] would make sense to have just one tableSaga
- [ ] would make sense to have a separate tableReducer
- [x ] no reason to remove the service
- [ x] why change standard mapStateToProps , mapDispatchToProps etc. names?
- [ x] please make full use of actions, use them to dispatch from component and saga and store all string consts for types
- [ x] connect will dispatch automatically, see https://blog.benestudio.co/5-ways-to-connect-redux-actions-50e4ad5ea096 #2

File: src/js/reducers/rootReducer.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { combineReducers } from 'redux'
+
+import table from "./tableReducer"
+
+export default () => combineReducers({
```
Comment (marina-igorevna-mishur) [inline src/js/reducers/rootReducer.js]: no need to wrap this in a lambda

File: src/js/index.js [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
 import ReactDOM from "react-dom"
+import { Provider } from "react-redux"
+import { createStore, applyMiddleware } from "redux"
+import createSagaMiddleware from "redux-saga"
 import App from "components/App.jsx"
+import createRootReducer from "reducers/rootReducer"
+import initSagas from "./initSagas"
 import "sass/main.scss"
 
+const rootReducer = createRootReducer()
```
Comment (marina-igorevna-mishur) [inline src/js/index.js]: no point in a create function here

File: src/js/index.js [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
+
+const sagaMiddleware = createSagaMiddleware()
+const store = createStore(
```
Comment (marina-igorevna-mishur) [inline src/js/index.js]: maybe you could move all of this to an `initRedux` file

File: src/js/components/Table.jsx [hunk 4]
```diff
@@ -1,39 +1,27 @@
 import React from "react"
-import TableDataService from "services/TableDataService"
+import { connect } from 'react-redux'
 import Row from "./Row.jsx"
+import PropTypes from "prop-types"
```
[the remark this answers is not in the document]
Comment (plantysnake) [reply src/js/components/Table.jsx]: Re-ordered

File: src/js/sagas/tableData.saga.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import * as actions from "reducers/actions"
+import TableDataService from "services/TableDataService"
+
+export function* TableDataSaga() {
+    yield takeEvery(actions.GET_HEADERS, fetchColumnsAsync)
+    yield takeEvery(actions.GET_TRIALS, fetchTrialsAsync)
+}
+
+function* fetchColumnsAsync() {
+    const data = yield call(() => {
```
[the remark this answers is not in the document]
Comment (plantysnake) [reply src/js/sagas/tableData.saga.js]: Re-edited without lambda

File: src/js/services/TableDataService.js [hunk 6]
```diff
@@ -1,10 +1,15 @@
 import http from "q-io/http"
-import { trials } from "utils/API"
+import { trials, headers } from "utils/API"
 
 function getTrialsInfo() {
     return http.read(trials).then(info => {
         return JSON.parse(info.toString())
     })
 }
+function getTableColumns() {
```
[the remark this answers is not in the document]
Comment (plantysnake) [reply src/js/services/TableDataService.js]: Merged into 1 func

File: src/js/components/Table.jsx [hunk 7]
```diff
@@ -1,39 +1,27 @@
...
 
-class Table extends React.Component {
-    constructor(props) {
-        super(props)
-        this.state = {
-            head: [],
-            trials: []
-        }
-    }
+const loadColumns = () => { return { type: GET_HEADERS} }
```
[the remark this answers is not in the document]
Comment (plantysnake) [reply src/js/components/Table.jsx]: Moved

File: src/js/reducers/rootReducer.js (hunk 1 above)
Comment (plantysnake) [reply src/js/reducers/rootReducer.js]: Removed

File: src/js/index.js (hunk 2 above)
Comment (plantysnake) [reply src/js/index.js]: Removed

File: src/js/index.js (hunk 3 above)
Comment (plantysnake) [reply src/js/index.js]: Moved into configureStore

**Pull request merged**